hera_pspec/plot.py

```diff
@@ -494,3 +496,176 @@ def delay_waterfall(uvp, blpairs, spw, pol, component='real', average_blpairs=Fa
+        
+        #Format k_para axis
+        x_axis = (uvp.get_dlys(0)*1e9).tolist()
+        if cosmo:
+            x_axis = uvp.get_kparas(0).tolist()
+        if fold:
+            x_axis.insert(0, 0)
+        ax.set_xlim((x_axis[0] / 2, x_axis[-1] / 2))
+        
+        #Format k_perp axis
+        #Calculate kpr indices, and stretch wedgeslices to length of baseline norms
+        bllen_indices = [int(bllen * 10) for bllen in bllens]
+        stretched_data = np.zeros((bllen_indices[-1], data.shape[-1]), dtype=np.float64)
+        j = 0
+        for i in range(len(bllen_indices)):
+            stretched_data[j:bllen_indices[i]] = data[i]
+            j = bllen_indices[i]
+        data = stretched_data[...]
+        
+        #Find kpr mid-indicies for the tickmarks
+        bllen_midindices = []
+        for i in range(len(bllen_indices)):
+            if i == 0:
+                bllen_midindices.append(bllen_indices[i] / 2.)
+            else:
+                bllen_midindices.append((bllen_indices[i] + bllen_indices[i - 1]) / 2.)
+        
+        #Setting y axis ticks and labels
+        ax.set_yticks(bllen_midindices)
+        ax.set_yticklabels(bllens, fontsize=10)
+        
+        #Plot the data
+        im = ax.imshow(
+            np.log10(np.abs(data)),
+            aspect='auto',
+            interpolation='nearest',
+            extent=[x_axis[0], x_axis[-1], bllen_indices[-1], 0])
+        
+        #Plot the horizon lines if requested
+        if horizon_lines:
+            horizons = [(bllen*u.m/c.c).to(u.ns).value for bllen in bllens]
+            #if cosmo:
+                #XXX still need to convert horizons to appropriate kpara values
+            j = 0
+            for i, (horizon, bllen) in enumerate(zip(horizons, bllens)):
+                x1, y1 = [horizon, horizon], [j, bllen_indices[i]]
+                x2, y2 = [-horizon, -horizon], [j, bllen_indices[i]]
+                ax.plot(x1, y1, x2, y2, color='#ffffff', linestyle='--', linewidth=1)
+                j = bllen_indices[i]
+        
+        #plot center line at k_para=0 if requested
+        if center_line:
+            ax.axvline(x=0, color='#000000', ls='--', lw=1)
+
+        ax.tick_params(axis='both', direction='inout')
+        ax.set_title(pol, fontsize=15)
+    
+    #add colorbar
+    cbar_ax = f.add_axes([0.9125, 0.25, 0.025, 0.5])
+    cbar = f.colorbar(im, cax=cbar_ax)
+    
+    #add axis labels
+    if cosmo:
+        f.text(0.5, 0.05, r'$k_{\parallel}\ [h\ Mpc^{-1}]$', fontsize=15, ha='center')            
+        f.text(0.07, 0.5, r'$k_{\perp}\ [\rm h\ Mpc^{-1}]$', fontsize=15, va='center', rotation='vertical')
+    else:
+        f.text(0.5, 0.05, r'$\tau\ [ns]$', fontsize=15, ha='center')
+        f.text(0.07, 0.5, r'Baseline Group $[m]$', fontsize=15, va='center', rotation='vertical')
+        
+    #add colorbar label
+    if cosmo_cbar:
+        cbar.set_label(r"$P(k_{\parallel})\ [mK^2h^{-3}Mpc^3]$", fontsize=15, ha='center')
```

Comment:
this should use the units found inside the `uvp` object. see other plotting functions for examples